DigitalLearningSolutions.Web/ViewModels/TrackingSystem/Delegates/EmailDelegates/EmailDelegatesViewModel.cs

```diff
@@ -2,83 +2,53 @@
 {
     using System;
     using System.Collections.Generic;
-    using System.ComponentModel.DataAnnotations;
     using System.Linq;
-    using DigitalLearningSolutions.Data.Helpers;
-    using DigitalLearningSolutions.Data.Models.CustomPrompts;
+    using DigitalLearningSolutions.Data.Models.SearchSortFilterPaginate;
     using DigitalLearningSolutions.Data.Models.User;
-    using DigitalLearningSolutions.Web.Helpers;
     using DigitalLearningSolutions.Web.ViewModels.Common.SearchablePage;
 
-    public class EmailDelegatesViewModel : BaseSearchablePageViewModel, IValidatableObject
+    public class EmailDelegatesViewModel : BaseSearchablePageViewModel<DelegateUserCard>
     {
-        public EmailDelegatesViewModel(string? existingFilterString) : base(
-            null,
-            1,
-            true,
-            DelegateSortByOptions.RegistrationDate.PropertyName,
-            GenericSortingHelper.Ascending,
-            existingFilterString,
-            int.MaxValue
-        ) { }
-
-        public EmailDelegatesViewModel() : this(null) { }
-
         public EmailDelegatesViewModel(
-            IEnumerable<DelegateUserCard> delegateUsers,
-            IEnumerable<(int id, string name)> jobGroups,
-            IEnumerable<CentreRegistrationPrompt> customPrompts,
-            string? existingFilterString,
+            SearchSortFilterPaginateResult<DelegateUserCard> result,
+            IEnumerable<FilterModel> availableFilters,
             bool selectAll = false
-        ) : this(existingFilterString)
+        ) : base(result, true, availableFilters)
         {
             Day = DateTime.Today.Day;
             Month = DateTime.Today.Month;
             Year = DateTime.Today.Year;
-            SetDelegates(delegateUsers, existingFilterString, selectAll);
-            SetFilters(jobGroups, customPrompts);
+            Delegates = result.ItemsToDisplay.Select(
+                delegateUser =>
+                {
+                    var delegateSelected = selectAll ||
+                                           SelectedDelegateIds != null && SelectedDelegateIds.Contains(delegateUser.Id);
+                    return new EmailDelegatesItemViewModel(delegateUser, delegateSelected);
+                }
+            );
+        }
+
+        public EmailDelegatesViewModel(
+            SearchSortFilterPaginateResult<DelegateUserCard> result,
+            IEnumerable<FilterModel> availableFilters,
+            EmailDelegatesFormData formData
+        ) : this(result, availableFilters)
+        {
+            SelectedDelegateIds = formData.SelectedDelegateIds;
+            Day = formData.Day;
+            Month = formData.Month;
+            Year = formData.Year;

```

Comment:
We have FormData as a base class of the ViewModel where we usually do this, but since we already have to have the BaseSearchable as our base class here, I had to do something a bit different. I figured this was the closest option.